### ISSUE TITLE
Template repository

### ISSUE THREAD

**Issue by nickmessing** [vuejs/vue-cli#221]
We have not only "official" vuejs templates, what about a repository for vuejs-templates?

**Comment by zigomir** [comment]
This would mean we'd need to change `vue-cli` and make it compatible for every git hosting vendor. Currently we use this convention that each template is a single repo with `template` directory in it.

Why do you think would be beneficial to have multiple templates in one repository? I understood your comment that you're suggesting this.

**Comment by posva** [comment]
I think he asking to have a list of nonofficial templates with the `list` command too or another command

**Comment by nickmessing** [comment]
@zigomir, @posva, I suggest making at least a web app, something like templates.vuejs.org where would be a list of all official and unofficial boilerplates for vue js, if @yyx990803 likes this idea I can make an app for that.

**Comment by nickmessing** [comment]
P.S. The problem I see right now is the lack of "vue+vuex+vue-router+whatever-people-use" template. I found a template only after making a PR to templates/webpack repo to add vuex and vue-router, at least an MD file would help (but I guess it's better to have a web app that tracks things like stars, last commit date and dependency versions)

**Comment by egoist** [comment]
@nickmessing eh, I think searching over github would find most of the vuejs templates

**Comment by nickmessing** [comment]
@egoist, isn't a curated list a better solution?

**Comment by LinusBorg** [comment]
Like this one? https://github.com/vuejs/awesome-vue#boilerplates

**Comment by nickmessing** [comment]
@LinusBorg, yes, that's exactly the place I started from, but:
1. There is no quick way to see if it's still maintained
2. There is no quick way to see if the dependencies are up-to-date
3. I can't see the popularity of each (at least github stars) without opening each of them and manually sorting.

So, for example, I start a project using a boilerplate, then I find out that it's built on Vue 1.

**Comment by LinusBorg** [comment]
A nicely curated list would be nice. If you have time to build something, the contribution would be welcome.

**Comment by nickmessing** [comment]
OFFTOPIC: @LinusBorg, check your profile description, "...hobby **fronend** hacker..."

**Comment by yyx990803** [comment]
@nickmessing that sounds like a great idea. Would be happy to feature it in the `vue-cli` readme or even help messages.

**Comment by egoist** [comment]
@nickmessing 

> 1. There is no quick way to see if it's still maintained
> 2. There is no quick way to see if the dependencies are up-to-date
> 3. I can't see the popularity of each (at least github stars) without opening each of them and manually sorting.

I guess your such web app is also suitable for Vue components 😅

**Comment by nickmessing** [comment]
@egoist, yep, right, I didn't use community components so I didn't even think about it :D.

**Comment by FadyMak** [comment]
@nickmessing I've done some work in the past on a similar idea for a template repository. It's quite incomplete but I could tackle this if you haven't already started work on it.

**Comment by nickmessing** [comment]
@FadyMak, I've finished back-end with Github API integration, nothing done on fronend yet

**Comment by Rayraegah** [comment]
well, I kinda use vue-cli on non-vue projects because it just works. What would be nice to have is a command like:

`$ vue list --templates 'http://<source>.feed'`

that could just display `user/repo` info to copy.
